docker-java-core/src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -600,7 +600,7 @@ public CreateContainerResponse exec() throws NotFoundException, ConflictExceptio
             containerNetwork.withAliases(aliases);
         }
 
-        if (containerNetwork != null) {
+        if (containerNetwork != null && hostConfig.getNetworkMode() != null) {
             networkingConfig = new NetworkingConfig()
                     .withEndpointsConfig(singletonMap(hostConfig.getNetworkMode(), containerNetwork));
```

Comment:
Exception was being thrown as we were putting:
```
EndpointsConfig: {
    null: containerNetwork
}
```
as networkMode is null by default. This can't be serialised by JSON so it throws.